brewcop.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python
+
+##############################################################
+#  Copyright 2019 Jim Garlick <garlick.jim@gmail.com>
+#  (c.f. COPYING)
+#
+#  This file is part of BREWCOP, a coffee pot monitor.
+#  For details, see https://github.com/garlick/brewcop.
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+##############################################################
+
+import urwid
+import time
+import subprocess
+import sys
+
+poll_period = 2
+
+
+class Scale:
+    path_helper = "/usr/local/bin/scale_query"
+
+    def __init__(self):
+        self.last_value = 0.0
+        self.tare_offset = 0.0
+
+    def tare(self):
+        self.tare_offset = self.last_value
+
+    def get_gross_weight(self):
+        out = subprocess.Popen(
+            [self.path_helper], stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+        )
+        stdout, stderr = out.communicate()
+        self.last_value = float(stdout) * 453.592
+        return self.last_value
+
+    def get_weight(self):
+        return self.get_gross_weight() - self.tare_offset
+
+    def get_weight_cached(self):
+        return self.last_value - self.tare_offset
+
+
+# Placeholder for "application logic"
+class Brewcop:
+    valid_modes = ["beans", "brew", "clean"]
+
+    def __init__(self):
+        self.mode = "beans"
+        self.weight = 0.0
+
+    def set_mode(self, mode):
```

Comment:
It is certainly not necessary, but if you wanted to, you could use the `@property` decorator [[1]](https://docs.python.org/2/library/functions.html#property)[[2]](https://stackoverflow.com/questions/6618002/using-property-versus-getters-and-setters) to trigger this logic when assignment is performed on mode (e.g., `brewcop.mode = "foo"` would fail).

Reply:
OK, cool, I will add that!

Reply:
Just pushed a commit that makes this change.  I like it!  Thanks for that.

(Needs a test with scale attached)